### ISSUE TITLE
Accessing FIXSession inside FIXStatusListener

### ISSUE THREAD

**Issue by jheusser** [jvirtanen/philadelphia#10]
In the example Initiator a `FIXStatusListener` is passed to the `FIXSession` as anonymous class (see [here](https://github.com/jvirtanen/philadelphia/blob/master/philadelphia-initiator/src/main/java/org/jvirtanen/philadelphia/initiator/Initiator.java#L33)). For more complicated `FIXStatusListeners` (or for multiple active `FIXSessions`) it's not convenient to pass it as anonymous class and thus I ended up with the repetitive and bit awkward pattern

```Java
BaseFixStatusListener statusListener = new BaseFixStatusListener();
FIXSession fixSession = new FIXSession(channel, fixConfig, inbound, statusListener);
statusListener.setFIXSession(fixSession);
```

Where the `BaseFixStatusListener` has a `setFIXSession` method. If you always want to access the `FIXSession` in a `FixStatusListener` it might make sense to add such a setter method to the interface, or similar.

**Comment by jvirtanen** [comment]
Yes, this is something I have been thinking about.

I think the cleanest approach would be to augment `FIXMessageListener` and `FIXStatusListener` methods to contain a `session` argument. An example:

```
void message(FIXSession session, FIXMessage message)
```

What do you think? @penberg, any thoughs?

**Comment by penberg** [comment]
@jvirtanen Sounds reasonable to me.

**Comment by jheusser** [comment]
@jvirtanen sounds good yes, also allows to potentially share a `FIXStatusListener` between multiple `FIXSessions`. I think the status listener has more immediate need for the session than the message listener, but probably makes sense for consistency to add the argument to both.

**Comment by jvirtanen** [comment]
@jheusser Good point again. I agree that we should do it only for `FIXStatusListener`, at least for now. Would you be interested in submitting a pull request?

**Comment by jheusser** [comment]
@jvirtanen Ok, will submit a PR adding a `FIXSession` parameter to each method in `FIXStatusListener`.

**Comment by jvirtanen** [comment]
Great, thanks! :+1:

**Comment by jheusser** [comment]
@jvirtanen Doing this now. Maybe it is also required in `FIXMessageListener` as the `MessageHandler` inside `FIXSession` is also passing down events to the session `FIXStatusListener`. Or I pass the session in via constructor in the `MessageHandler`. You decide what's best.

**Comment by jvirtanen** [comment]
You don't need to pass the session to `MessageListener`; as a non-static inner class you can refer to the current session as `FIXSession.this` in `MessageListener`.

**Comment by jheusser** [comment]
Right it's not static, duh! Thanks

@jvirtanen  Another issue my IDE is just explaining about in `philadelphia-perf-test`. The package `org.philadelphia.perf` does not match the package name used within the java class files (which is `org.jvirtanen.philadelphia.perf`).

**Comment by jvirtanen** [comment]
Great find! I myself use Vim, so I didn't notice the mismatch. Would you like to open another pull request to fix that?

**Comment by jheusser** [comment]
Will do

**Comment by jvirtanen** [comment]
Great, thanks! :+1:

**Comment by jvirtanen** [comment]
Closed by #13.
